hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/hollow/RandomHollowAccount.java

```diff
@@ -0,0 +1,106 @@
+import static com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HUNDRED_HBARS;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.ACCOUNT_DELETED;
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.bdd.spec.HapiSpecOperation;
+import com.hedera.services.bdd.spec.infrastructure.HapiSpecRegistry;
+import com.hedera.services.bdd.spec.infrastructure.OpProvider;
+import com.hedera.services.bdd.spec.infrastructure.providers.names.RegistrySourcedNameProvider;
+import com.hedera.services.bdd.spec.queries.meta.HapiGetTxnRecord;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import java.util.Optional;
+
+public class RandomHollowAccount implements OpProvider {
+    // Added to hollow account names to differentiate them from the keys created for them
+    public static final String ACCOUNT_SUFFIX = "#";
+    public static final String KEY_PREFIX = "Fuzz#";
+    public static final int DEFAULT_CEILING_NUM = 100;
+    public static final long INITIAL_BALANCE = 1_000_000_000L;
+    private int ceilingNum = DEFAULT_CEILING_NUM;
+    private final HapiSpecRegistry registry;
+
+    private final RegistrySourcedNameProvider<Key> keys;
+    private final RegistrySourcedNameProvider<AccountID> accounts;
+
+    public RandomHollowAccount(
+            HapiSpecRegistry registry,
+            RegistrySourcedNameProvider<Key> keys,
+            RegistrySourcedNameProvider<AccountID> accounts) {
+        this.registry = registry;
+        this.keys = keys;
+        this.accounts = accounts;
+    }
+
+    public RandomHollowAccount ceiling(int n) {
+        ceilingNum = n;
+        return this;
+    }
+
+    @Override
+    public Optional<HapiSpecOperation> get() {
+        // doubling ceilingNum as keys are also saved in accounts registry when account is created
+        if (accounts.numPresent() >= ceilingNum * 2) {
+            return Optional.empty();
+        }
+
+        return randomKey().map(this::generateHollowAccount);
+    }
+
+    private Optional<String> randomKey() {
+        return keys.getQualifying()
+                .filter(k -> !k.endsWith(ACCOUNT_SUFFIX))
+                .filter(k -> k.startsWith(KEY_PREFIX))
+                .filter(k -> !registry.hasAccountId(k + ACCOUNT_SUFFIX));
+    }
+
+    private HapiSpecOperation generateHollowAccount(String keyName) {
+        return withOpContext((spec, opLog) -> {
+            final var evmAddress = getEvmAddress(keyName);
+            final var op = cryptoTransfer(tinyBarsFromTo(GENESIS, evmAddress, ONE_HUNDRED_HBARS))
+                    .hasKnownStatusFrom(standardOutcomesAnd(ACCOUNT_DELETED))
+                    .via("LAZY_CREATE");
```

Comment:
Is it safe to use the same `"LAZY_CREATE"` string for each such `cryptoTransfer()`, instead of like `"LAZY_CREATE" + newAtomicLongField.getAndIncrement()`?


If 3 concurrent `RandomHollowAccount` operations being run, it seems some of them might get the wrong response from the following `hapiGetTxnRecord` query.

Reply:
Hmm, yeah that does not look safe, I've updated the code as recommended.